components/dialog/dialog-mixin.js

```diff
@@ -322,9 +322,11 @@ export const DialogMixin = superclass => class extends RtlMixin(superclass) {
 			await this._updateSize();
 			this._state = 'showing';
 
-			// edge case: no children in footer slot were selected, try again after one redraw
+			// edge case: no children were focused, try again after one redraw
 			const activeElement = getComposedActiveElement();
-			if (!activeElement || !isComposedAncestor(dialog, activeElement)) {
+			if (!activeElement
+			|| !isComposedAncestor(dialog, activeElement)
+			|| !activeElement.classList.contains('focus-visible')) {
```

Comment:
This entire conditional is becoming a pretty nasty hack, but the last expression seems to be required since the ARIA dialog autofocusing the close button without the focus ring. May or may not be the optimal solution